From 8ece74954e1cf0348b46bc818fb72c4dcf413c72 Mon Sep 17 00:00:00 2001
From: Brennan Lamoreaux <blamoreaux@vmware.com>
Date: Wed, 21 May 2025 07:58:41 +0000
Subject: [PATCH] adding option to set description field of module

[Guru: Renegrated patch for kpatch version 0.9.10
Also merged multi line description support changes with
this patch]
Signed-off-by: Guruswamy Basavaiah <guruswamy.basavaiah@broadcom.com>
---
 kpatch-build/kpatch-build | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/kpatch-build/kpatch-build b/kpatch-build/kpatch-build
index bc112fc..1869518 100755
--- a/kpatch-build/kpatch-build
+++ b/kpatch-build/kpatch-build
@@ -719,6 +719,7 @@ usage() {
 	echo "		-r, --sourcerpm         Specify kernel source RPM" >&2
 	echo "		-s, --sourcedir         Specify kernel source directory" >&2
 	echo "		-c, --config            Specify kernel config file" >&2
+	echo "		-D, --description       Specify module description file" >&2
 	echo "		-v, --vmlinux           Specify original vmlinux" >&2
 	echo "		-j, --jobs              Specify the number of make jobs" >&2
 	echo "		-t, --target            Specify custom kernel build targets" >&2
@@ -741,7 +742,7 @@ if ! command -v gawk &> /dev/null; then
 	die "gawk not installed"
 fi
 
-options="$(getopt -o ha:r:s:c:v:j:t:n:o:dR -l "help,archversion:,sourcerpm:,sourcedir:,config:,vmlinux:,jobs:,target:,name:,output:,oot-module:,oot-module-src:,debug,skip-gcc-check,skip-compiler-check,skip-cleanup,non-replace,version" -- "$@")" || die "getopt failed"
+options="$(getopt -o ha:r:s:c:D:v:j:t:n:o:dR -l "help,archversion:,sourcerpm:,sourcedir:,config:,description:,vmlinux:,jobs:,target:,name:,output:,oot-module:,oot-module-src:,debug,skip-gcc-check,skip-compiler-check,skip-cleanup,non-replace,version" -- "$@")" || die "getopt failed"
 
 eval set -- "$options"
 
@@ -770,6 +771,11 @@ while [[ $# -gt 0 ]]; do
 		CONFIGFILE="$(readlink -f "$2")"
 		shift
 		;;
+	-D|--description)
+		[[ ! -f "$2" ]] && die "description file '$2' not found"
+		DESC_FILE="$(readlink -f "$2")"
+		shift
+		;;
 	-v|--vmlinux)
 		[[ ! -f "$2" ]] && die "vmlinux file '$2' not found"
 		VMLINUX="$(readlink -f "$2")"
@@ -1506,6 +1512,12 @@ KPATCH_LDFLAGS="$KPATCH_LDFLAGS" \
 CROSS_COMPILE="$CROSS_COMPILE"
 save_env
 
+if [[ $DESC_FILE ]]; then
+	MOD_DESC="$(tr '\n' ' ' < $DESC_FILE)"
+	MOD_DESC="${MOD_DESC%%[[:space:]]}"
+	echo "MODULE_DESCRIPTION(\"$MOD_DESC\");" >> $TEMPDIR/patch/livepatch-patch-hook.c || die
+fi
+
 make "${MAKEVARS[@]}" 2>&1 | logger || die
 
 if [[ "$USE_KLP" -eq 1 ]]; then
-- 
2.40.4

